runtime/lua/vim/shared.lua

```diff
@@ -1139,4 +1139,77 @@ function vim._defer_require(root, mod)
   })
 end
 
+--- @nodoc
+--- @class vim.context.mods
+--- @field buf? integer
+--- @field hide? boolean
+--- @field horizontal? boolean
+--- @field keepalt? boolean
+--- @field keepjumps? boolean
+--- @field keepmarks? boolean
+--- @field keeppatterns? boolean
+--- @field lockmarks? boolean
+--- @field noautocmd? boolean
+--- @field options? table<string, any>
+--- @field sandbox? boolean
+--- @field win? integer
+
+--- Wrapper over the `vim._with_c` function to restore editor state that is difficult to do from
+--- the C code.
+---
+--- @param context vim.context.mods
+function vim._with(context, f, ...)
```

Comment:
we use `ctx` in most places 
```suggestion
function vim._with(ctx, f, ...)
```